src/common/grpc.rs

```
@@ -412,12 +412,34 @@ where
         self.encode_message(status)
     }
 
-    fn board_pwm(&mut self, _message: &[u8]) -> Result<(), GrpcError> {
-        Err(GrpcError::RpcUnimplemented)
+    fn board_pwm(&mut self, message: &[u8]) -> Result<(), GrpcError> {
+        let req = component::board::v1::PwmRequest::decode(message)
+            .map_err(|_| GrpcError::RpcInvalidArgument)?;
+        let board = match self.robot.lock().unwrap().get_board_by_name(req.name) {
+            Some(b) => b,
+            None => return Err(GrpcError::RpcUnavailable),
+        };
+        let pin: i32 = req.pin.parse::<i32>().unwrap();
```

could you return an error instead here?